src/main/java/org/benetech/servicenet/service/impl/DocumentUploadServiceImpl.java

```diff
@@ -25,22 +35,34 @@
 
     private final Logger log = LoggerFactory.getLogger(DocumentUploadServiceImpl.class);
 
-    private final DocumentUploadRepository documentUploadRepository;
+    @Autowired
+    private DocumentUploadRepository documentUploadRepository;
 
-    private final DocumentUploadMapper documentUploadMapper;
+    @Autowired
+    private DocumentUploadMapper documentUploadMapper;
 
-    public DocumentUploadServiceImpl(DocumentUploadRepository documentUploadRepository,
-                                     DocumentUploadMapper documentUploadMapper) {
-        this.documentUploadRepository = documentUploadRepository;
-        this.documentUploadMapper = documentUploadMapper;
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private MongoDbService mongoDbService;
+
+    @Override
+    public DocumentUploadDTO uploadFile(MultipartFile file, String delimiter) throws IllegalArgumentException, IOException {
+
+        String parsedDocumentId = null;
+        try {
+            AbstractFileConverter converter = FileConverterFactory.getConverter(file, delimiter);
+            parsedDocumentId = mongoDbService.saveParsedDocument(converter.convert(file));
+        } catch (IOException ex) {
+            log.error(ex.getMessage(), ex);
```

Comment:
I think we want to fail at this point, not just log the error?

Reply:
You're right, I'll change it.